content/blog/deploying-mongodb-application-on-hpe-greenlake-for-containers.md

```diff
@@ -0,0 +1,948 @@
+hpe-logzio                Active   71d
+hpe-metering              Active   71d
+hpe-nodesvc               Active   71d
+hpe-operations            Active   50d
+hpe-secure                Active   71d
+hpe-security              Active   71d
+hpe-snow                  Active   71d
+hpe-storage               Active   71d
+hpe-system                Active   71d
+hpe-templates-compute     Active   71d
+hpecp                     Active   71d
+hpecp-bootstrap           Active   71d
+hpecp-falco               Active   71d
+kd-apps                   Active   71d
+kd-mlops                  Active   71d
+kube-node-lease           Active   71d
+kube-public               Active   71d
+kube-system               Active   71d
+kubernetes-dashboard      Active   71d
+mongo                     Active   6s
+opsramp-agent             Active   71d
+
+Switch the new namespace in the current context
+
+$ kubectl config set-context --current --namespace=mongo
+Context "caas-dev-3-hpe-hpe-ashish-kumar@hpe.com" modified.
+
+$ kubectl config view --minify --output 'jsonpath={..namespace}'; echo
+mongo
+
+$ kubectl get pods
+No resources found in mongo namespace.
+```
+
+### Step-6: Deploy MongoDB
+
+```shellsession
+$ kubectl create -f install-mongo.yaml
+service/mongo created
+statefulset.apps/mongo created
+
+$ cat install-mongo.yaml
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: mongo
+  labels:
+    name: mongo
+    hpecp.hpe.com/hpecp-internal-gateway: "true" # Expose the service on ECP Gateway
+spec:
+  ports:
+  - protocol: TCP
+    port: 27017
+    targetPort: 27017
+  selector:
+    role: mongo
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: mongo
+spec:
+  selector:
+    matchLabels:
+      app: mongo
+  serviceName: "mongo"
+  # Number of initial MongoDB pods to deploy - default is 1 if not set
+  replicas: 3
+  podManagementPolicy: Parallel
+  template:
+    metadata:
+      labels:
+        app: mongo
+    spec:
+      terminationGracePeriodSeconds: 10
+      containers:
+        - name: mongo
+          image: mongo
+          env:
+          - name: POD_IP_ADDRESS
+            valueFrom:
+              fieldRef:
+                fieldPath: status.podIP
+          command:
+          - "mongod"
+          - "--bind_ip_all"
+          - "--replSet"
+          - rs0
+          - "--oplogSize"
+          - "128"
+          ports:
+            - containerPort: 27017
+          volumeMounts:
+            - name: mongo-persistent-storage
+              mountPath: /data/db
+        - name: mongo-sidecar
+          image: cvallance/mongo-k8s-sidecar
+          env:
+            - name: MONGO_SIDECAR_POD_LABELS
+              value: "app=mongo"
+  volumeClaimTemplates:
+    - metadata:
+        name: mongo-persistent-storage
+      spec:
+        accessModes: [ "ReadWriteOnce" ]
+        storageClassName: gl-sbc-glhcnimblestor
+        resources:
+          requests:
+            storage: 50Gi
+```
+
+### Step-7: Validate MongoDB Deployment
```

Comment:
Same comment here. Also, "Deployment" should be "deployment"

Reply:
done